src/components/ZeplinPanel.tsx

```diff
@@ -1,140 +1,142 @@
-import React, { Component } from "react";
+import React, { useEffect, useState, useCallback } from "react";
 import { styled } from "@storybook/theming";
 
-import HeaderButtons from "./HeaderButtons";
 import { getZeplinResource } from "../utils/api";
+import HeaderButtons from "./HeaderButtons";
 
-interface ZeplinPanelProps {
-    zeplinLink: string;
+interface ZeplinkLink {
+    name: string;
+    link: string;
 }
 
-interface ZeplinPanelState {
-    zeplinData: {
-        name: string;
-        image: {
-            original_url: string;
-        };
-    };
-    loading: boolean;
-    error: string;
-    zoomLevel: number;
+interface ZeplinPanelProps {
+    zeplinLink: ZeplinkLink[];
 }
 
-class ZeplinPanel extends Component<ZeplinPanelProps, ZeplinPanelState> {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            zeplinData: null,
-            loading: true,
-            error: null,
-            zoomLevel: 1,
-        };
-    }
-
-    fetchZeplinResource = async () => {
-        const { zeplinLink } = this.props;
-
-        if (!zeplinLink) {
+const ZeplinPanel: React.FC<ZeplinPanelProps> = ({ zeplinLink }) => {
+    const [selectedLink, setSelectedLink] = useState("");
+    const [zeplinData, setZeplinData] = useState(null);
+    const [zoomLevel, setZoomLevel] = useState(1);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
```

Comment:
Let's combine all of these states into one using `useReducer`. Then we don't need to call consecutive setState functions which causes multiple rerenders.

```js
const [state, setState] = useReducer(
    (currentState: AppState, newState: Partial<AppState>): AppState => ({ ...currentState, ...newState }),
    initialAppState
);
```

Reply:
That sounds like an idea, although if you are concerned regarding multiple rendering, React will batch the setState calls if they're done within a React event handler, same as would happen if you were to call `this.setState({...})` multiple times in a class based component.

https://github.com/facebook/react/issues/14259#issuecomment-439632622

Reply:
That's good. Thanks for the info 👍. But I'd still prefer using `useReducer` if it works for you too.

Reply:
Definitely!